test/test.contracts.js

```diff
@@ -0,0 +1,312 @@
+          publicKeyHash: 'tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx',
+          seed: '',
+          storeType: 1
+      };
+
+describe('Smart Contract Operations', async () => {
+    let sandbox = null, contract;
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+        sandbox1 = sinon.createSandbox();
+        sandbox
+            .stub(jsonfile, 'readFileSync')
+            .withArgs(confFile)
+            .returns(testconfig);
+        contract = new ContractClass.Contracts();
+    });
+    afterEach(() => {
+        sandbox.restore()
+    });
+
+    context('deploy-contract-with-args', async () => {
+        it('must call sendContractOriginationOperation function with applied status', async () => {
+            const conseiljs = require(CONSEIL_JS);
+            stubReadFileSync = sandbox
+                                .stub(fs, 'readFileSync')
+                                .withArgs('./MichelsonCodeFile.tz')
+                                .returns(CONTRACT_CODE);
+
+            stubKeys = sandbox
+                        .stub(contract, 'getKeys')
+                        .withArgs(BOOTSTRAPPED_ACCOUNT)
+                        .returns( {
+                            pk: 'edpkuBknW28nW72KG6RoHtYW7p12T6GKc7nAbwYX5m8Wd9sDVC9yav',
+                            sk: 'edskRuR1azSfboG86YPTyxrQgosh5zChf5bVDmptqLTb5EuXAm9rsnDYfTKhq7rDQujdn5WWzwUMeV3agaZ6J2vPQT58jJAJPi',
+                            pkh: 'tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx',
+                            seed: '',
+                            storeType: 1
+                        });
+
+            stubConseil = sandbox
+                            .stub(conseiljs.TezosNodeWriter, 'sendContractOriginationOperation')
+                            .withArgs(testconfig.provider, keystore, 0, undefined,
+                                      100000, '', 10000, 500000, 
+                                      CONTRACT_CODE, CONTRACT_INITIAL_STORAGE, conseiljs.TezosParameterFormat.Michelson)
+                            .returns(sendContractOriginationOperation.applied);
+
+            stubLoggerWarn = sandbox.stub(Logger, 'warn');
+            stubLoggerInfo = sandbox.stub(Logger, 'info');
+            stubLoggerError = sandbox.stub(Logger, 'error');        
+
+            ophash = sendContractOriginationOperation.applied.results.contents[0].metadata.operation_result.originated_contracts;
+
+            stubAddNewContract = sandbox
+                                .stub(contract, 'addNewContract')
+                                .withArgs(CONTRACT_DEPLOY_LABEL, CONTRACT_ADDRESS, BOOTSTRAPPED_ACCOUNT, testconfig.provider);
+                                
+            await contract.deployContract([CONTRACT_DEPLOY_LABEL, './MichelsonCodeFile.tz', CONTRACT_INITIAL_STORAGE, BOOTSTRAPPED_ACCOUNT]);
+            sinon.assert.calledOnce(stubReadFileSync);
+            sinon.assert.calledOnce(stubConseil);
+            sinon.assert.calledOnce(stubAddNewContract);
```

Comment:
wherever a function returns a value, you can assert if returned value is same or not.

Reply:
yes, will do that, I forgot to include this. Other than this please review the code.